static/app/components/searchQueryBuilder/tokens/combobox.tsx

```diff
@@ -56,9 +56,9 @@ type SearchQueryBuilderComboboxProps<T extends SelectOptionOrSectionWithKey<stri
   onCustomValueCommitted: (value: string) => void;
   /**
    * Called when the user selects an option from the dropdown.
-   * Passes the value of the selected item.
+   * Passes the selected option.
    */
-  onOptionSelected: (value: string) => void;
```

Comment:
Needed to pass the full option object here so that we could do different things when a query option is selected vs a filter key.